### PR TITLE
Wildcard labels

### PR DESCRIPTION
**Pull request by geolich**
Hey, this allows to select arbitrary nodes based on a regex.

It doesnt follow XPath official spec, but it could be very useful to read arbitrary nodes based on regex.

### CODE REVIEW

File: xtract-core/src/main/scala/com/lucidchart/open/xtract/XPath.scala [hunk 1]
```diff
@@ -19,6 +20,11 @@ case class RecursiveXPathNode(key: String) extends XPathNode {
   override def toString = s"//$key"
 }
 
+case class WildCardXPathNode(regex: Regex) extends XPathNode {
+  def apply(xml: NodeSeq): NodeSeq = (xml \\ "_").filter(node => node.label.matches(regex.regex))
```
Comment (tmccombs) [inline xtract-core/src/main/scala/com/lucidchart/open/xtract/XPath.scala:24]: Should this be `xml \ "_"`? or is the intention to search recursively? or maybe we should have both `\?` and `\\?` for non-recursive and recursive?

**Review by tmccombs — CHANGES_REQUESTED** [review]
Thanks for the contribution! I definitely think this is a useful addition. I just have one concern, (see other comment)

File: xtract-core/src/main/scala/com/lucidchart/open/xtract/XPath.scala (hunk 1 above)
Comment (geolich) [reply xtract-core/src/main/scala/com/lucidchart/open/xtract/XPath.scala:24]: Hey, yes makes sense. I will add both

**Comment by geolich** [comment]
I now added both recursive and non-recursive

**Pull request merged**